sklearn/metrics/_plot/tests/test_plot_roc_curve.py

```diff
@@ -168,3 +181,51 @@ def test_default_labels(pyplot, roc_auc, estimator_name,
     disp = RocCurveDisplay(fpr=fpr, tpr=tpr, roc_auc=roc_auc,
                            estimator_name=estimator_name).plot()
     assert disp.line_.get_label() == expected_label
+
+
+@pytest.mark.parametrize(
+    "response_method", ["predict_proba", "decision_function"]
+)
+def test_plot_roc_curve_pos_label(pyplot, response_method):
+    # check that we can provide the positive label and display the proper
+    # statistics
+    X, y = load_breast_cancer(return_X_y=True)
+    # create an highly imbalanced
+    idx_positive = np.flatnonzero(y == 1)
+    idx_negative = np.flatnonzero(y == 0)
+    idx_selected = np.hstack([idx_negative, idx_positive[:25]])
+    X, y = X[idx_selected], y[idx_selected]
+    X, y = shuffle(X, y, random_state=42)
+    # only use 2 features to make the problem even harder
+    X = X[:, :2]
+    y = np.array(
+        ["cancer" if c == 1 else "not cancer" for c in y], dtype=object
+    )
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, stratify=y, random_state=0,
+    )
+
+    classifier = LogisticRegression()
+    classifier.fit(X_train, y_train)
+
+    # sanity check to be sure the positive class is classes_[0] and that we
+    # are betrayed by the class imbalance
+    assert classifier.classes_.tolist() == ["cancer", "not cancer"]
+
+    disp = plot_roc_curve(
+        classifier, X_test, y_test, pos_label="cancer",
+        response_method=response_method
+    )
+
+    roc_auc_limit = 0.96
+    assert disp.roc_auc < roc_auc_limit
+    assert -np.trapz(disp.tpr, disp.fpr) < roc_auc_limit
```

Comment:
This is not OK for the ROC curve. We should have an exact match

```suggestion
    assert np.trapz(disp.tpr, disp.fpr) == pytest.approx(roc_auc_limit)
```

Reply:
And I expect this to fail because we have the bug that should fix the computation of `roc_auc_score`: https://github.com/scikit-learn/scikit-learn/pull/17594